casper/src/main/scala/coop/rchain/casper/api/GraphGenerator.scala

```diff
@@ -158,6 +148,47 @@ object GraphzGenerator {
       node = Map("width" -> "0", "height" -> "0", "margin" -> "0.03", "fontsize" -> "8")
     )
 
+  private def drawParentDependencies[G[_]: Applicative](
+      g: Graphz[G],
+      validators: List[(String, ValidatorsBlocks)]
+  ): G[Unit] =
+    validators
+      .map(_._2)
+      .toList
+      .flatMap(_.values.toList)
+      .traverse {
+        case ValidatorBlock(blockHash, parentsHashes, _) => {
+          parentsHashes
+            .traverse(p => g.edge(blockHash, p, constraint = Some(false)))
+        }
+      }
+      .as(())
+
+  private def drawJustificationDottedLines[G[_]: Applicative](
+      g: Graphz[G],
+      validators: Map[String, ValidatorsBlocks]
```

Comment:
Similar story here, you're using only the values so no need to pass the whole map with keys